### ISSUE TITLE
Support OpenQASM 3.0

### ISSUE THREAD

**Issue by balopat** [quantumlib/Cirq#3518]
Currently we only support OpenQASM 2.0, we should start to look into what is required for exporting and importing to and from the new version of [OpenQASM 3.0](https://qiskit.github.io/openqasm/)

**What is the urgency from your perspective for this issue? Is it blocking important work?**
P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by vtomole** [comment]
I will close this issue as a duplicate of https://github.com/quantumlib/Cirq/issues/4509 even though this was the first one because the linked issue has more comments.
